arangod/CMakeLists.txt

```diff
@@ -839,7 +839,7 @@ add_library(arango_rocksdb STATIC
   ${ADDITIONAL_LIB_ARANGO_ROCKSDB_SOURCES}
 )
 
-if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+if (CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_CLANG)
   target_compile_options(arango_rocksdb PUBLIC "-maes")
```

Comment:
I think this change is good. 
After it is merged, we can check whether we can roll back part of the changes in enterprise/c7dea6bfba5eea5c1bad581e492e721a7184bd91, which disabled AES intrinsics on MacOS.